### PR TITLE
Added more registration fields

### LINKED ISSUE #1867: Toevoegen van initiator velden voor StUF-ZDS en ZGW-APIs
Initiator velden toevoegen aan StUF-ZGW en ZGW-API:

Inhoud veld | Veldtype | Registratiecode
-- | -- | --
Initiator
Handelsnaam | tekstveld | Initiator   > Handelsnaam
KvK nummer | getalveld | Initiator   > KvK
Vestigingsnummer | getalveld | Initiator   > Vestigingsnummer
Voorletters | tekstveld | Initiator   > Voorletters
Voornaam | tekstveld | Initiator   > Voornamen
Tussenvoegsel | tekstveld | Initiator   > Tussenvoegsel
Geslachtsnaam | tekstveld | Initiator   > Geslachtsnaam
Geboortedatum | Datumveld | Initiator   > Geboortedatum
Straat | tekstveld | Initiator   > Straat
Huisnummer | getalveld | Initiator   > Huisnummer
Huisletter | tekstveld | Initiator   > Huisletter
Huisnummer toevoeging | tekstveld | Initiator   > Huisnummertoevoeging
Postcode | Postcode | Initiator   > Postcode
Woonplaats | Tekstveld | Initiator   > Woonplaats
Telefoonnummer | Telefoonnummer | Initiator   > Telefoonnummer
Mailadres | E-mail | Initiator   > Email
Locatie (voor gerelateerde locatie bij   zaak)
Straatnaam | tekstveld | Locatie   > Straat
Huisnummer | getalveld | Locatie   > Huisnummer
Huisletter | tekstveld | Locatie   > Huisletter
Huisnummertoevoeging | tekstveld | Locatie   > Huisnummertoevoeging
Postcode | postcode | Locatie   > Postcode
Plaats | tekstveld | Locatie   > Plaats

</body>

</html>

### PR DESCRIPTION
**Pull request by joeribekker**
Fixes #1867

I was not able to add location info as I didn't see any separate location resource/XML-schema. Address information is added though.

### CODE REVIEW

**Review by joeribekker — COMMENTED** [review]
@dariomory I see you removed the `kvk` attribute from the ZGW test: https://github.com/open-formulieren/open-forms/pull/2025/commits/82f2288e63dae3ce34f795e15a8b83abd4eebcff#diff-857464e519bdf16c980dc00fc37c7db08396ba2b86225132f7d5542ae6671621R407-L383

It is however present in the StUF-ZDS version of the test, so I'd like to keep them the same.

Can we make the KvK a different number from the Vestigingsnummer? There is some logic that says "if there is a Vestigingsnummer" it overrides the KvK field

So please change those tests, and add 1 more (sub)test that only sends the kvk-number (so no vestigingsnummer, but the kvk number should be shown in the vestigingsnumber field in SOAP and REST

File: src/openforms/registrations/contrib/zgw_apis/tests/test_backend.py [hunk 1]
```diff
@@ -187,6 +194,20 @@ def test_submission_with_zgw_backend(self, m):
                         "attribute": RegistrationAttribute.initiator_geboortedatum,
                     },
                 },
+                {
+                    "key": "geslachtsaanduiding",
+                    "type": "textfield",
+                    "registration": {
+                        "attribute": RegistrationAttribute.initiator_geslachtsaanduiding,
+                    },
+                },
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/zgw_apis/tests/test_backend.py:203]: (in the future) this could be a preset field maybe

File: src/openforms/registrations/contrib/zgw_apis/plugin.py [hunk 2]
```diff
@@ -143,6 +177,19 @@ def register_submission(
...
+
+        betrokkene_identificatie = rol_data.get("betrokkeneIdentificatie", {})
+        kvk = betrokkene_identificatie.pop("_kvk", None)
+
+        if kvk:
+            rol_data["betrokkeneType"] = "vestiging"
+            # Fall back on KvK-number if no vestigingsnumber.
+            betrokkene_identificatie["vestigingsNummer"] = betrokkene_identificatie.get(
+                "vestigingsNummer", kvk
+            )
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/zgw_apis/plugin.py:189]: I think this doesn't catch the case where the field has an empty string for value - then there will not be a fallback to `kvk`.

File: src/openforms/registrations/contrib/zgw_apis/tests/test_backend.py [hunk 3]
```diff
@@ -304,6 +328,351 @@ def test_submission_with_zgw_backend(self, m):
...
+        )
+        self.assertEqual(
+            relate_attachment_body["zaak"], "https://zaken.nl/api/v1/zaken/1"
+        )
+        self.assertEqual(
+            relate_attachment_body["informatieobject"],
+            "https://documenten.nl/api/v1/enkelvoudiginformatieobjecten/2",
+        )
+
+    def test_submission_with_zgw_backend_with_vestiging_initiator_kvk_only(self, m):
```
Comment (sergei-maertens) [inline src/openforms/registrations/contrib/zgw_apis/tests/test_backend.py:546]: I'm missing a test case where an optional `vestigingsNummer` field is present with an empty string as value.

File: src/stuf/stuf_zds/templates/stuf_zds/soap/creeerZaak.xml [hunk 4]
```diff
@@ -57,25 +57,55 @@
...
                             {% if initiator.voornamen %}<BG:voornamen>{{ initiator.voornamen }}</BG:voornamen>{% endif %}
-{#                            <BG:geslachtsaanduiding>?</BG:geslachtsaanduiding>#}
+                            {% if initiator.geslachtsaanduiding %}<BG:geslachtsaanduiding>{{ initiator.geslachtsaanduiding }}</BG:geslachtsaanduiding>{% endif %}
                             {% if initiator.geboortedatum %}<BG:geboortedatum StUF:indOnvolledigeDatum="{{ initiator.geboortedatum.indicator }}">{{ initiator.geboortedatum.value }}</BG:geboortedatum>{% endif %}
+
+                            {% if initiator.verblijfsadres.postcode or initiator.verblijfsadres.woonplaatsNaam %}
+                                {% include "stuf_zds/soap/includes/verblijfsadres.xml" %}
+                            {% endif %}
                         </ZKN:natuurlijkPersoon>
                     {% elif initiator.kvk %}
```
Comment (sergei-maertens) [inline src/stuf/stuf_zds/templates/stuf_zds/soap/creeerZaak.xml:85]: what if there's no `initiator.kvk`, but there is `initiator.vestigingsNummer`?

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
I think the newly added tests are doing too much and aren't focused on just the extra StUF-ZDS/ZGW attributes, but since they're passing I'll let it slide.

**Review by sergei-maertens — APPROVED** [review]
I'll rebase this into commits that can be cherry-picked for backport!

**Comment by sergei-maertens** [comment]
Backported in bfbcbe2b974c31c911b95d1727e620a3bff1f853 and f4d5b8125a02d7ecd86dad8034323dc3b01879e1
